packages/aws-cdk-lib/aws-ecr/lib/repository.ts

```diff
@@ -544,6 +546,27 @@ export class Repository extends RepositoryBase {
       throw new Error('"repositoryArn" is a late-bound value, and therefore "repositoryName" is required. Use `fromRepositoryAttributes` instead');
     }
 
+    const arnRegex = /^arn:(\w+):ecr:([a-z]+-[a-z]+-\d{1}):(\d{12}):repository\/([A-Za-z]+.*)$/;
```

Comment:
instructions unclear 😂 I meant that the `validateRepoArn` function can be a function that lives inside `fromRepositoryArn`. Like this:

```ts
public static fromRepositoryArn(repositoryArn: string) {
  if (Token.isUnresolved(repositoryArn)) {
    throw new Error();  
  }

  validateRepositoryArn();

  const repositoryName = repositoryArn.split('/').slice(1).join('/');

  class Import extends RepositoryBase {}

  return new Import(scope, id, {
    environmentFromArn: repositoryArn,
  });

  function validateRepositoryArn() {
    // all your validations here
  }
}
```